test/03_reduction/reduction_test.cpp

```diff
@@ -296,9 +369,9 @@ namespace hiptensor
                                                                      resource->deviceC().get(),
                                                                      &descC,
                                                                      modeC.data(),
-                                                                     resource->deviceC().get(),
-                                                                     &descC,
-                                                                     modeC.data(),
+                                                                     resource->deviceD().get(),
+                                                                     &descD,
+                                                                     modeD.data(),
```

Comment:
Memory leak below if you don't release the work pointer

Reply:
Can we allocate with the HipResource and use the unique_ptr manage the work memory here?